src/node/internal/internal_dns.ts

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+// Copyright Joyent and Node contributors. All rights reserved. MIT license.
+
+import {
+  sendDnsRequest,
+  validateAnswer,
+  normalizeMx,
+  normalizeCname,
+  normalizeCaa,
+  normalizePtr,
+  normalizeNaptr,
+  normalizeNs,
+  normalizeSoa,
+  normalizeTxt,
+  normalizeSrv,
+  type MX,
+  type CAA,
+  type NAPTR,
+  type SOA,
+  type SRV,
+  type TTLResponse,
+} from 'node-internal:internal_dns_client';
+import { DnsError } from 'node-internal:internal_errors';
+import { validateString } from 'node-internal:validators';
+import * as errorCodes from 'node-internal:internal_dns_constants';
+
+export function getServers(): string[] {
+  throw new Error('Not implemented');
+}
+
+export function lookup(): void {
+  throw new Error('Not implemented');
+}
+
+export function lookupService(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolve(): void {
+  throw new Error('Not implemented');
+}
+
+export function resolve4(
+  name: string,
+  options?: { ttl?: boolean }
+): Promise<(string | TTLResponse)[]> {
+  validateString(name, 'name');
+
+  // The following change is done to comply with Node.js behavior
+  let ttl = false;
+  if (options != null && typeof options === 'object') {
+    ttl = Boolean(options.ttl);
+  }
```

Comment:
Should be able to simplify to just....

```suggestion
  const ttl = !!options?.ttl;
```

Reply:
I have how easy JS enables you to write safe JS code that looks unsafe.